### PR TITLE
[BBC-7102] Using Storybook Swatch and Type Scaling Components

### PR DESCRIPTION
**Pull request by gabrielagabriel**
## Description

https://www.loom.com/share/64767267ac394e3ba3786a19e01d66c3

## What has been done

- Replace _Colors_ and _Font_ export for the Storybook components
- `Spacing` still an script
- Rename typography  spec from `.md` to `mdx`
- Star to group components by "function" (new taxonomy to promote discoverability). 

## Things to consider

Notice that I have not re-written any documentation yet.. polish on all that will come later

## How it was tested

UT

### CODE REVIEW

File: src/_utils/rideAxis/story.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { color, font } from '../branding'
 import { RideAxis } from './index'
 import readme from './specifications/index.md'
 
-const stories = storiesOf('Internal|RideAxis', module)
+const stories = storiesOf('Widgets|RideAxis', module)
```
Comment (grosbouddha) [inline src/_utils/rideAxis/story.tsx:10]: this should not be here ?

File: src/typography/body/story.tsx [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
+import { boolean, text } from '@storybook/addon-knobs'
 import { storiesOf } from '@storybook/react'
 
 import { color } from '../../_utils/branding'
 import { BaseSection as Section } from '../../layout/section/baseSection'
 import { TextBody } from './index'
 
-const stories = storiesOf('Brand|typography', module)
-stories.addDecorator(withKnobs)
+const stories = storiesOf('Design Tokens|Typography/', module)
```
Comment (pith) [inline src/typography/body/story.tsx:9]:
```suggestion
const stories = storiesOf('Design Tokens|Typography', module)
```

Comment (gabrielagabriel) [reply src/typography/body/story.tsx:9]: The `/` is used to create a folder

File: src/subHeader/story.tsx [hunk 3]
```diff
@@ -1,17 +1,12 @@
...
 
 import { BaseSection as Section } from '../layout/section/baseSection'
 import { SubHeader } from '../subHeader'
-import spec from './specifications/subHeader.md'
 
-const stories = storiesOf('Widgets|SubHeader', module)
-stories.addDecorator(withKnobs)
-const specTemplateFn = () => <SubHeader>SubHeader content</SubHeader>
-stories.add('Specifications', specTemplateFn, { readme: spec })
+const stories = storiesOf('Text|', module)
```
Comment (grosbouddha) [inline src/subHeader/story.tsx:7]: Why not keep Text|SubHeader like other stories ?

File: src/typography/body/story.tsx (hunk 2 above)
Comment (pith) [inline src/typography/body/story.tsx:9]: Is the `/` expected in `'Design Tokens|Typography/'`? Could we share constants for these strings?

File: src/_utils/rideAxis/story.tsx (hunk 1 above)
Comment (gabrielagabriel) [reply src/_utils/rideAxis/story.tsx:10]: I'm moving the widgets by group of function like "Design Tokens", `RideAxis` where all alone so I move it back to Widgets, I can do it on other PR, though.. no problem

File: src/typography/titleStrong/story.tsx [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
+import { boolean, text } from '@storybook/addon-knobs'
 import { storiesOf } from '@storybook/react'
 
 import { color } from '../../_utils/branding'
 import { BaseSection as Section } from '../../layout/section/baseSection'
 import { TextTitleStrong } from './index'
 
-const stories = storiesOf('Brand|typography', module)
-stories.addDecorator(withKnobs)
+const stories = storiesOf('Design Tokens|Typography/', module)
```
Comment (grosbouddha) [inline src/typography/titleStrong/story.tsx:9]: Do we need the 'Design Tokens' level ? could we just go for Typography ?
It's not clear what's a design token (the meaning might vary depending on which medium article you read :/)

File: src/typography/body/story.tsx (hunk 2 above)
Comment (gabrielagabriel) [reply src/typography/body/story.tsx:9]: Yes.. it creates a "folder" structure. The way we declare hierarchy will change on 6.0, But good point we may think about this optimisation

File: src/subHeader/story.tsx (hunk 3 above)
Comment (gabrielagabriel) [reply src/subHeader/story.tsx:7]: 'cos "Default" can be declared direct as story, no need to create another level of hierarchy

File: src/typography/titleStrong/story.tsx (hunk 4 above)
Comment (gabrielagabriel) [reply src/typography/titleStrong/story.tsx:9]: A design token is an 'atom' basically. We can't do any design without colors, type, scalling... 
We can redefine a new taxonomy. I'm going to open a JIRA for that.

**Pull request merged**